lib/utils.py

```diff
@@ -108,6 +107,21 @@ def set_extras_n_fix_units(
 
     args["resource_provides_quoted"] = [fixquote(x) for x in args["resource_provides"]]
 
+    # if the user defined the usage_model on the command line,
+    # we need to use their definition of usage_model, not ours
+    # also, we need to define the sites as 'Fermigrid' if they want to run locally
+    for r in args["resource_provides_quoted"]:
+        print(f"r is: {r}")
+        if "usage_model" in r:
+            args["usage_model"] = ""
+        if r == 'usage_model="OPPORTUNISTIC,DEDICATED"':
+            args["site"] = "Fermigrid"
```

Comment:
In this case, don't we still also want to set `args["usage_model"]` to `OPPORTUNISTIC,DEDICATED`?  Or is the Fermigrid specification enough to steer the jobs to Fermigrid?

Reply:
From the submit file changes, I suspect that `Fermigrid` may be enough, but please confirm that.

Reply:
There are two arg parameters, "--usage_model" and "--resource_provides", both of which could have a usage_model definition in them. (Note that the 'resource_provides' parameter could have OTHER definitions as well.) The 'resource_provides' parameter is left empty if the user doesn't define it. 'usage_model' is filled with the default value usage_model="OPPORTUNISTIC,DEDICATED,OFFSITE" if the user doesn't define it through the --offsite or --onsite parameters.

If someone sets the 'resource_provides' parameter to a usage_model in the command line, such as in "--resource-provides=usage_model=OPPORTUNISTIC,DEDICATED", we don't want the code to override their choice of usage_model. That is why I set the parameter "usage_model" to the empty string in line 116. This only happens if they have explicitly defined a usage_model in their command line.

Hopefully this makes sense.

Reply:
That is correct.